### PR TITLE
TECH-378: Remove old IAM resources

### PR DESCRIPTION
**Pull request by magreenbaum**
## Description 

This is to remove old IAM resources that do not have the region name suffixes. Moved new resources in `iam-new.tf` into `main.tf` to consolidate.  

See also changes to cloudwatch log group in comments. Using later aws provider versions has changed arn of cloudwatch logs. 

## Test

Applied in dev to confirm appropriate resources are destroyed and that the cloudwatch changes are accurate (note that this means bumping aws provider version to v3 in app repo using this module). 

## To Do 

Once cutover to the new IAM instance profile is completed for all RMQ launch configurations, update source ref of rmq module in CMW to the hash of this commit. Also bump aws provider version to >3. This will be at least another 1.5 to 2 weeks out.

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -100,7 +93,7 @@ data "aws_iam_policy_document" "policy_permissions_doc" {
     ]
     resources = [
       aws_cloudwatch_log_group.log_group.arn,
-      "${aws_cloudwatch_log_group.log_group.arn}/*"
+      "${aws_cloudwatch_log_group.log_group.arn}:*"
```
Comment (magreenbaum) [inline main.tf]: Bumping aws provider to latest revealed a bugfix that was done here which removed the `:*` from the cloudwatch logs arn: https://github.com/hashicorp/terraform-provider-aws/pull/14214

Also noting that the arn format of log groups is:
Log group | arn:aws:logs:region:account-id:log-group:log_group_name
Log stream | arn:aws:logs:region:account-id:log-group:log_group_name:log-stream:log-stream-name
https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/iam-access-control-overview-cwl.html

Planning to bump CMW terraform aws provider to latest v3 which will have the arn bugfix. 

Tested in dev: 
```
            "Resource": [
                "arn:aws:logs:<region>:<account>:log-group:rabbitmq-<app_name>-dev:*:*",
                "arn:aws:logs:<region>:<account>:log-group:rabbitmq-<app_name>-dev:*"
```
Logs show up in CW.

File: main.tf [hunk 2]
```diff
@@ -99,8 +92,8 @@ data "aws_iam_policy_document" "policy_permissions_doc" {
       "logs:PutLogEvents"
     ]
     resources = [
-      aws_cloudwatch_log_group.log_group.arn,
-      "${aws_cloudwatch_log_group.log_group.arn}/*"
+      "${aws_cloudwatch_log_group.log_group.arn}:*",
+      "${aws_cloudwatch_log_group.log_group.arn}:*:*"
```
Comment (magreenbaum) [inline main.tf:96]: Bumping aws provider to latest revealed a bugfix that was done here which removed the `:*` from the cloudwatch logs arn: https://github.com/hashicorp/terraform-provider-aws/pull/14214

Also noting that the arn format of log group and log stream is:
`Log group | arn:aws:logs:region:account-id:log-group:log_group_name`
`Log stream | arn:aws:logs:region:account-id:log-group:log_group_name:log-stream:log-stream-name`
https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/iam-access-control-overview-cwl.html

Planning to bump CMW terraform aws provider to latest v3 which will have the arn bugfix. Required for these changes.

Tested in dev: 
```
            "Resource": [
                "arn:aws:logs:<region>:<account>:log-group:rabbitmq-<app_name>-dev:*:*",
                "arn:aws:logs:<region>:<account>:log-group:rabbitmq-<app_name>-dev:*"
```
Logs show up in CW.

Comment (rraub) [reply main.tf:96]: good catch!

**Review by rraub — APPROVED** [review]
Looks good!

**Pull request merged**